packages/vm/src/instance.rs

```diff
@@ -97,9 +97,11 @@ where
         let fe = FunctionEnv::new(&mut store, {
             let e = Environment::new(backend.api, gas_limit);
             if print_debug {
-                e.set_debug_handler(Some(Rc::new(RefCell::new(|msg: &str, _gas_remaining| {
-                    eprintln!("{msg}");
-                }))))
+                e.set_debug_handler(Some(Rc::new(RefCell::new(
+                    |msg: &str, _gas_remaining: DebugInfo<'_>| {
```

Comment:
I think this explicit type is what I was missing when trying with two lifetimes.
And the compiler error is not very helpful, because it actually says it found `DebugInfo<'_>`, but expected `DebugInfo<'b>`. I'm guessing the annonymous lifetime is a different one.